### PR TITLE
vanillify setDimensions

### PR DESCRIPTION
**Pull request by amtur8**

### CODE REVIEW

File: slick/slick.js [hunk 1]
```diff
@@ -1937,40 +1937,43 @@
...
+            slideTrack.style['height'] = Math.ceil(_.outerHeight(slides[0]) * slideTrack.getElementsByClassName('slick-slide').length) + 'px';
         }
 
-        var offset = _.$slides.first().outerWidth(true) - _.$slides.first().width();
-        if (_.options.variableWidth === false) _.$slideTrack.children('.slick-slide').width(_.slideWidth - offset);
+        var offset = _.outerWidth(slides[0]) - _.width(slides[0]);
+        if (_.options.variableWidth === false) {
+            Array.prototype.forEach.call(slideTrack.getElementsByClassName('slick-slide'), function(element) {
+                element.style['width'] = _.slideWidth - offset + 'px';
+            });
```
Comment (pieplu) [inline slick/slick.js]: Un petit `_.queryAll()` ? :p

**Comment by pieplu** [comment]
It's looks good :)
Maybe resolve conflict and ,on the same time, use `queryAll` :p

File: slick/slick.js [hunk 2]
```diff
@@ -1945,40 +1948,43 @@ Issues: http://github.com/kenwheeler/slick/issues
...
+			slideTrack.style['height'] = Math.ceil(_.outerHeight(slides[0]) * slideTrack.getElementsByClassName('slick-slide').length) + 'px';
 		}
 
-		var offset = _.$slides.first().outerWidth(true) - _.$slides.first().width();
-		if (_.options.variableWidth === false) _.$slideTrack.children('.slick-slide').width(_.slideWidth - offset);
+		var offset = _.outerWidth(slides[0]) - _.width(slides[0]);
+		if (_.options.variableWidth === false) {
+			Array.prototype.forEach.call(slideTrack.getElementsByClassName('slick-slide'), function(element) {
+				element.style['width'] = _.slideWidth - offset + 'px';
+			});
```
Comment (pieplu) [inline slick/slick.js:1996]: Why not at this place too?

`_.queryAll('.slick-slide', slideTrack).forEach(...)`

Comment (amtur8) [inline slick/slick.js:1996]: didn't see it. I will change this.

**Comment by pieplu** [comment]
Update this branch to develop trigger test :) @amtur8

**Pull request merged**